nav2_rviz_plugins/src/costmap_cost_tool.cpp

```diff
@@ -106,34 +116,11 @@ void CostmapCostTool::callCostService(float x, float y)
   request->use_footprint = false;
 
   // Call local costmap service
-  if (local_cost_client_->wait_for_service(std::chrono::seconds(1))) {
```

Comment:
I think these should be reverted, the invoke shared would block the thread which would make all of rviz seize up until the service client got a response.

Reply:
I think this is still true, while we don't block up the thread anymore since there's no `spin`, we immediately `.get()` the future, which is probably iinvalid since its not yet completed. I think the callback method where we call and triggers a callback when completed to log is still proper.